### PR TITLE
Shop homepage redesign

### LINKED ISSUE #196: Redesign Shop: Landing Page
## Description
There's an older version of Landing Page on `./pages/shop/sections/CategoriesProducts` and `./pages/shop/Homepage` files. You could try refactoring it to [this Figma Design](https://www.figma.com/design/2r8LJbRj7TEXZ7obQvVeb8/SpaceYaTech-Website-Re-design?node-id=10310-6555&t=BKQWYyXN1n1b57Kn-0) for the Shop Landing Page.

## Tasks

- [ ] Implement Hero carousel, filter/search functionality and FAQ; in the same format as the existing components from the `/events` and `/` pages.
- [ ] Create a reusable shop item card 
- [ ] Make the UI responsive on mobile

### PR DESCRIPTION
**Pull request by bugswritter**
[Have you read the contributing guidelines ?](https://github.com/SpaceyaTech/SYT-Web-Redesign/blob/Dev/docs/CONTRIBUTING.md)

# What is the purpose of your _pull request_?

- [ ] Bug fix
- [x] New feature
- Fixes #196 
# Proposed changes
This pull request introduces a complete redesign of the landing page for our shop, incorporating several new features and improvements.
1. Carousel for Hero Section and Featured Products
2. Product Search and Filter Functionality.
3. Mobile Responsiveness.

# Warning

Please read these points carefully and answer honestly with an `X`
into all the boxes. Example : [X]

Before submitting a _pull request_ make sure you have:

- [x] Read the guidelines for contributing.
- [ ] Wrote some tests.
- [x] Respected the linting guidelines.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,9 @@
     "class-variance-authority": "^0.7.0",
     "clsx": "^2.1.1",
     "date-fns": "^2.30.0",
+    "embla-carousel": "^8.1.5",
+    "embla-carousel-autoplay": "^8.1.5",
+    "embla-carousel-react": "^8.1.5",
```
Comment (sonylomo) [inline package.json:33]: PR #207 has an alternative implementation of the carousel without the additional packages.

Comment (sonylomo) [inline package.json]: PR #207  has an alternative implementation of the carousel without the additional packages. Otherwise, you've done a good job.

**Review by sonylomo — CHANGES_REQUESTED** [review]
PR #207 has an alternative implementation of the carousel without the additional packages.

File: package.json (hunk 1 above)
Comment (sonylomo) [reply package.json]: The PR has been merged so you can view the changes on the `Dev` branch.

Comment (bugswritter) [reply package.json]: Okay, I have removed the packages now using the existing implemetation

**Review by sonylomo — APPROVED** [review]

**Pull request merged**